troubleshooting/debugging.md

```diff
@@ -161,6 +165,57 @@ And Fluent Bit logs show this SIGTERM message, then this indicates that Fluent B
 #### caught signal SIGSEGV
 
 If you see a `SIGSEGV` in your Fluent Bit logs, then unfortunately you have encountered a bug! This means that Fluent Bit crashed due to a segmentation fault, an internal memory access issue. Please cut us an issue on GitHub, and consider checking out the [SIGSEGV debugging section](#segfaults-and-crashes-sigsegv) of this guide. 
+#### Log4j TCP Appender Write Failure
+
+Many customers use [log4j](https://logging.apache.org/log4j/2.x/manual/cloud.html) to emit logs, and use the TCP appender to write to a [Fluent Bit TCP input](https://docs.fluentbit.io/manual/pipeline/inputs/tcp). ECS customers can follow this [ECS Log Collection Tutorial](https://github.com/aws-samples/amazon-ecs-firelens-examples/tree/mainline/examples/fluent-bit/ecs-log-collection). 
+
+Unfortunately, log4j can sometimes fail to write to the TCP socket and you may get an error like:
+
+```
+2022-09-05 14:18:49,694 Log4j2-TF-1-AsyncLogger[AsyncContext@6fdb1f78]-1 ERROR Unable to write to stream TCP:localhost:5170 for appender ApplicationTcp org.apache.logging.log4j.core.appender.AppenderLoggingException: Error writing to TCP:localhost:5170 after reestablishing connection for localhost/127.0.0.1:5170
+```
+
+Or:
+
+```
+Caused by: java.net.SocketException: Connection reset by peer (Write failed)
+```
+
+If this occurs, your first step is to check if it is simply a misconfiguration. Ensure you have configured a Fluent Bit TCP input for the same port that log4j is using. 
+
+Unfortunately, AWS has received occasional reports that log4j TCP appender will fail randomly in a previously working setup. AWS is tracking reports of this problem here: https://github.com/aws/aws-for-fluent-bit/issues/294
+
+While we can not prevent connection failures from occurring, we have found several mitigations.
+
+##### Mitigation 1: Enable Workers for all outputs
+
+AWS testing has found that TCP input throughput is maximized when Fluent Bit is performing well. If you enable at least one worker for all outputs, this ensures your outputs each have their own thread(s). This frees up the main Fluent Bit scheduler/input thread to focus on the inputs. 
```

Comment:
This makes sense. Good solution.

Reply:
Do you think you should mention that workers is set to 1 by default so people don't get confused?

Reply:
yea